dbcon/mysql/ha_mcs_execplan.cpp

```diff
@@ -6408,9 +6424,15 @@ void parse_item(Item* item, vector<Item_field*>& field_vec, bool& hasNonSupportI
 
     case Item::REF_ITEM:
     {
+      Item* thisBlockItem = directRefItemOrNull(item);
+      if (thisBlockItem)
+      {
+        parse_item(thisBlockItem, field_vec, hasNonSupportItem, parseInfo, gwi);
+        break;
+      }
```

Comment:
I am not sure why a call to `parse_item()` is added here. `real_item()` should be pulled before the first switch statement.